core/src/main/java/org/infinispan/statetransfer/StateTransferManagerImpl.java

```diff
@@ -90,7 +90,7 @@
    @Inject IracManager iracManager;
    @Inject IracVersionGenerator iracVersionGenerator;
 
-   private final CountDownLatch initialStateTransferComplete = new CountDownLatch(1);
+   private final CompletableFuture<Void> initialStateTransferComplete = new CompletableFuture<>();
```

Comment:
hmm, isn't a `CompletableFuture` kind of overkill?

Reply:
The `CountDownLatch.getCount()` javadoc says it should only used for testing, so I wanted to get rid of it, and `CompletableFuture.isDone()` sounded cleaner than `await(0, MILLISECONDS)`

Reply:
sometimes I have the feeling we are "abusing" the `CompletableFuture` when a simples `Semaphore`, `CountDownLatch`, `boolean` is enough :rofl: